src/cdk/layout/media-matcher.ts

```diff
@@ -0,0 +1,67 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import {Injectable} from '@angular/core';
+import {Platform} from '@angular/cdk/platform';
+
+/**
+ * Global registry for all dynamically-created, injected style tags.
+ */
+let styleElementForWebkitCompatibility: {[key: string]: HTMLStyleElement} = {};
+
+@Injectable()
+export class MediaMatcher {
+  /** The internal matchMedia method to return back a MediaQueryList like object. */
+  private _matchMedia: (query: string) => MediaQueryList;
+
+  constructor(private platform: Platform) {
+    this._matchMedia = this.platform.isBrowser ?
+      (<any>window).matchMedia.bind((<any>window)) :
+      (query: string) => <MediaQueryList>{
+        matches: query === 'all' || query === '',
+        media: query,
+        addListener: () => {},
+        removeListener: () => {}
+      };
+  }
+
+  /**
+   * Confirms the layout engine will trigger for the selector query provided and returns the
+   * MediaQueryList for the query provided.
+   */
+  matchMedia(query: string): MediaQueryList {
+    if (this.platform.WEBKIT) {
+      createEmptyStyleRule(query);
+    }
+    return this._matchMedia(query);
+  }
+}
+
+/**
+ * For Webkit engines that only trigger the MediaQueryListListener when there is at least one CSS
+ * selector for the respective media query.
+ */
+function createEmptyStyleRule(query: string) {
+  if (!styleElementForWebkitCompatibility[query]) {
+    try {
```

Comment:
Why the `try-catch`? Especially when the `catch` still logs the error

Reply:
This is actually a hold over from @ThomasBurleson 's original implementation.  My understanding is the `try-catch` is intended to allow the service to fail gracefully and not cause an entire halt.

Reply:
My stance is that any `try-catch` need a comment explaining which error it is catching and why it's something that has to be caught instead of prevented in the first place

@ThomasBurleson do you remember if there was a specific reason for using `try` around creating a style element?